### DISCUSSION TITLE
Dynamic API route with query string

### DISCUSSION THREAD

**Discussion by akshay-nm** [vercel/next.js#37181] [Help]
if I have a route `/api/users/[id]`
and I send a request to `/api/users/1?abc='xyz'`
Then do both values (`id` and `abc`) end up in `req.query`?

**Comment by icyJoseph** [comment]
Yes.

It also handles overriding, favouring `[id]` from the url path, so:

```
GET http://localhost:3000/api/2001?tomato=2&id=2002
```

Still yields `req.query.id` equal to `2001`.
